### PR TITLE
Added accounts.freedom.tm authentication

### PR DESCRIPTION
**Pull request by eshleebien**

### CODE REVIEW

File: app/Http/Controllers/Auth/AuthController.php [hunk 1]
```diff
@@ -30,36 +23,69 @@ class AuthController extends Controller
...
-        ]);
+        if (!Input::get('access_token')) {
+
+            return redirect('/au');
+        }
+
+        $access_token = Input::get('access_token');
+        $state = Input::get('state');
+
+        $client = new Client(['base_uri' => 'http://api.accounts.freedom.tm']);
```
Comment (wildtron) [inline app/Http/Controllers/Auth/AuthController.php]: put to config

File: app/Http/Controllers/Auth/AuthController.php [hunk 2]
```diff
@@ -30,36 +23,69 @@ class AuthController extends Controller
...
+        $data = [
+                'service' => 'gototm',
+                'redirect_uri' => 'http://goto.tm/au/callback',
+                'response_type' => 'code',
+                'roles' => 'profile,email,partner',
+                'state' => '/'
+            ];
+
+        $param = http_build_query($data);
+        $url = 'http://api.accounts.freedom.tm/auth?';
```
Comment (wildtron) [inline app/Http/Controllers/Auth/AuthController.php]: put to config

File: app/Http/Controllers/Auth/AuthController.php [hunk 3]
```diff
@@ -30,36 +23,69 @@ class AuthController extends Controller
...
-     *
-     * @param  array  $data
-     * @return User
-     */
-    protected function create(array $data)
+    public function getIndex()
+    {
+        $data = [
+                'service' => 'gototm',
+                'redirect_uri' => 'http://goto.tm/au/callback',
```
Comment (wildtron) [inline app/Http/Controllers/Auth/AuthController.php]: put to config

File: resources/views/home.blade.php [hunk 4]
```diff
@@ -5,27 +5,30 @@
...
+                <br><br>
+                <div class="content">
+                    <form action="redirect" method="POST">
+                        <input type="hidden" name="_token" value="{{ csrf_token() }}">
+                        <label>URL</label>
+                        <input type="text" name="long_url"/>
+                        <label>New URL</label>
+                        <input type="text" name="short_url" placeholder="http://goto.tm/"/>
+                        <input type="submit"/>
+                        <div class="g-recaptcha" data-sitekey="6LfWsgoTAAAAABb44i-KdeWKVXEe8ghQat6hIel2"></div>
```
Comment (Jhoem) [inline resources/views/home.blade.php]: put to config?

File: resources/views/success.blade.php [hunk 5]
```diff
@@ -5,17 +5,20 @@
...
-                <a href="/">create more</a>
+        @include('partials.header')
+        <div class="main">
+            <div class="container">
+                <span class="title">Hooray!</span>
+                <br>
+                <div class="content">
+                    <h1>
+                        You can now use
+                        <a target="_blank" href="{{ config('app.url') }}/{{ $data['short_url'] }}"/>
```
Comment (robertsambuena) [inline resources/views/success.blade.php:16]: ng-href

File: resources/views/success.blade.php [hunk 6]
```diff
@@ -5,17 +5,20 @@
...
+                            {{ config('app.url') }}/{{ $data['short_url'] }}
```
Comment (robertsambuena) [inline resources/views/success.blade.php:17]: ng-bind

File: resources/views/success.blade.php (hunk 5 above)
Comment (eshleebien) [inline resources/views/success.blade.php:16]: there's no angular here

Comment (robertsambuena) [inline resources/views/success.blade.php:16]: ahh I see
**RESOLVED**

File: resources/views/success.blade.php (hunk 6 above)
Comment (robertsambuena) [inline resources/views/success.blade.php:17]: **RESOLVED**

**Comment by Jhoem** [comment]
:up:

**Pull request merged**